core/src/main/java/org/infinispan/statetransfer/StateConsumerImpl.java

```diff
@@ -358,7 +358,10 @@ public CompletionStage<CompletionStage<Void>> onTopologyUpdate(CacheTopology cac
          stateTransferLock.acquireExclusiveTopologyLock();
          try {
             this.cacheTopology = cacheTopology;
-            distributionManager.setCacheTopology(cacheTopology);
+            // Protect from mocks
+            if (distributionManager instanceof DistributionManagerImpl dm) {
```

Comment:
Should we add a new internal interface solely for the DistributionManager instead similar to what we did for InternalDataContainer?